modules/social_features/social_content_block/src/ContentBuilder.php

```diff
@@ -284,4 +286,110 @@ public static function processBlockForm(array $element, FormStateInterface $form
     return $element;
   }
 
+  /**
+   * Sorting and range logic by specific case.
+   *
+   * @param \Drupal\Core\Database\Query\SelectInterface $query
+   *   The query.
+   * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+   *   The entity type that is being queried.
+   * @param string $sort_by
+   *   The type of sorting that should happen.
+   */
+  protected function sortBy(SelectInterface $query, EntityTypeInterface $entity_type, string $sort_by) : void {
+    // Define a lower limit for popular content so that content with a large
+    // amount of comments/votes is not popular forever.
+    // Sorry cool kids, your time's up.
+    $popularity_time_start = strtotime('-90 days');
+
+    // Provide some values that are often used in the query.
+    $entity_type_id = $entity_type->id();
+    $entity_id_key = $entity_type->getKey('id');
+
+    switch ($sort_by) {
+      // Creates a join to select the number of comments for a given entity
+      // in a recent timeframe and use that for sorting.
+      case 'most_commented':
+        if ($entity_type_id === 'group') {
+          $query->leftJoin('post__field_recipient_group', 'pfrg', "base_table.${entity_id_key} = pfrg.field_recipient_group_target_id");
+          $query->leftJoin('comment_field_data', 'cfd', "(base_table.${entity_id_key} = cfd.entity_id AND cfd.entity_type=:entity_type) OR (pfrg.entity_id = cfd.entity_id AND cfd.entity_type='post')", ['entity_type' => $entity_type_id]);
+        }
+        // Otherwise only check direct votes.
+        elseif ($entity_type_id === 'node') {
+          $query->leftJoin('comment_field_data', 'cfd', "base_table.${entity_id_key} = cfd.entity_id AND cfd.entity_type=:entity_type", ['entity_type' => $entity_type_id]);
+        }
+
+        $query->addExpression('COUNT(cfd.cid)', 'comment_count');
+        $query
+          ->condition('cfd.status', 1, '=')
+          ->condition('cfd.created', $popularity_time_start, '>')
+          ->groupBy("base_table.${entity_id_key}")
+          ->orderBy('comment_count', 'DESC');
+        break;
+
+      // Creates a join to select the number of likes for a given entity in a
+      // recent timeframe and use that for sorting.
+      case 'most_liked':
+        // For groups also check likes on posts in groups. This does not (yet) take into account likes on comments on posts or likes on other group content entities.
+        if ($entity_type_id === 'group') {
+          $query->leftJoin('post__field_recipient_group', 'pfrg', "base_table.${entity_id_key} = pfrg.field_recipient_group_target_id");
+          $query->leftJoin('votingapi_vote', 'vv', "(base_table.${entity_id_key} = vv.entity_id AND vv.entity_type=:entity_type) OR (pfrg.entity_id = vv.entity_id AND vv.entity_type='post')", ['entity_type' => $entity_type_id]);
+        }
+        // Otherwise only check direct votes.
+        elseif ($entity_type_id === 'node') {
```

Comment:
We're not doing anything node specific here so we should just make this else. This also ensures that the expression after the if/else doesn't break when this is used with e.g. posts ;-)

Reply:
Fixed.